### PR TITLE
Detect older tokenizer

### PR DESCRIPTION
**Pull request by Wovchena**
Ticket 156944

### CODE REVIEW

**Comment by moslex** [comment]
Can we add bw test for it as well? Like always checking "-1" release.

**Comment by Wovchena** [comment]
How should the test look like? The model never fails

**Comment by andrei-kochin** [comment]
> How should the test look like? The model never fails

Assert that for old IR we get old special_token handling?

**Review by apaniukov — APPROVED** [review]

File: src/cpp/src/tokenizer.cpp [hunk 1]
```diff
@@ -128,6 +132,11 @@ class Tokenizer::TokenizerImpl {
 
         auto device = "CPU"; // currently openvino_tokenizer supports only CPU
         auto ov_tokenizer = core.read_model(tokenizer_path / "openvino_tokenizer.xml");
+        if (ov_tokenizer->get_rt_info().count("openvino_tokenizers_version") == 1) {
+            m_older_than_24_5 = false;
```
[the remark this answers is not in the document]
Comment (Wovchena) [reply src/cpp/src/tokenizer.cpp]: Updated

File: src/cpp/src/tokenizer.cpp [hunk 2]
```diff
@@ -92,6 +93,9 @@ class Tokenizer::TokenizerImpl {
         if (add_special_tokens == m_add_special_tokens) {
             return;
         }
+        if (m_older_than_24_5) {
+            return;
```
[the remark this answers is not in the document]
Comment (Wovchena) [reply src/cpp/src/tokenizer.cpp:100]: Added

File: src/cpp/src/tokenizer.cpp [hunk 3]
```diff
@@ -92,6 +93,12 @@ class Tokenizer::TokenizerImpl {
         if (add_special_tokens == m_add_special_tokens) {
             return;
         }
+        if (m_older_than_24_5) {
+            // Changing add_special_tokens at runtime was introduced in
+            // 24.5. Older tokenizers sill allow manipulating their
```
Comment (ilya-lavrenov) [inline src/cpp/src/tokenizer.cpp:98]:
```suggestion
            // 24.5. Older tokenizers still allow manipulating their
```

**Pull request merged**